tests/neuro_san/internals/run_context/langchain/llms/test_llm_factory.py

```diff
@@ -10,154 +10,27 @@
-                request_timeout=self.get_value_or_env(config, "request_timeout", None, async_client),
-                max_retries=self.get_value_or_env(config, "max_retries", None, async_client),
-
-                presence_penalty=config.get("presence_penalty"),
-                frequency_penalty=config.get("frequency_penalty"),
-                seed=config.get("seed"),
-                logprobs=config.get("logprobs"),
-                top_logprobs=config.get("top_logprobs"),
-                logit_bias=config.get("logit_bias"),
-                streaming=True,  # streaming is always on. Without it token counting will not work.
-                n=1,  # n is always 1.  neuro-san will only ever consider one chat completion.
-                top_p=config.get("top_p"),
-                max_tokens=config.get("max_tokens"),  # This is always for output
-                tiktoken_model_name=config.get("tiktoken_model_name"),
-                stop=config.get("stop"),
-
-                # The following three parameters are for reasoning models only.
-                reasoning=config.get("reasoning"),
-                reasoning_effort=config.get("reasoning_effort"),
-                verbosity=config.get("verbosity"),
-
-                # If omitted, this defaults to the global verbose value,
-                # accessible via langchain_core.globals.get_verbose():
-                # https://github.com/langchain-ai/langchain/blob/master/libs/core/langchain_core/globals.py#L53
-                #
-                # However, accessing the global verbose value during concurrent initialization
-                # can trigger the following warning:
-                #
-                # UserWarning: Importing verbose from langchain root module is no longer supported.
-                # Please use langchain.globals.set_verbose() / langchain.globals.get_verbose() instead.
-                # old_verbose = langchain.verbose
-                #
-                # To prevent this, we explicitly set verbose=False here (which matches the default
-                # global verbose value) so that the warning is never triggered.
-                verbose=False,
-
-                # Set stream_usage to True in order to get token counting chunks.
-                stream_usage=True
-            )
-
-        elif chat_class is None:
-            raise ValueError(f"Class name {chat_class} for model_name {model_name} is unspecified.")
-        else:
-            raise ValueError(f"Class {chat_class} for model_name {model_name} is unrecognized.")
 
-        # Return the LlmResources with the client_policy that was created.
-        # That might be None, and that's OK.
-        return LangChainLlmResources(llm, client_policy=client_policy)
+        class_to_llm_policy_type: Dict[str, Type[LlmPolicy]] = {
+            "test-openai": OpenAILlmPolicy
+        }
+        super().__init__(class_to_llm_policy_type)
```

Comment:
Now we are simply passing the class name -> LlmPolicy type lookup table into the constructor.
Et voila.